src/code/z_actor.c

```diff
@@ -456,232 +469,236 @@ void Target_SetColors(TargetContext* targetCtx, Actor* actor, s32 type, PlayStat
-                            var2 = 0.15f;
-                        } else {
-                            var2 = ((entry->unkC - 120.0f) * 0.001f) + 0.15f;
-                        }
+                if (entry->distance < 500.0f) {
+                    s32 lockOnIndex;
+
+                    if (entry->distance <= 120.0f) {
+                        var2 = 0.15f;
+                    } else {
+                        var2 = ((entry->distance - 120.0f) * 0.001f) + 0.15f;
+                    }
 
-                        Matrix_Translate(entry->pos.x, entry->pos.y, 0.0f, MTXMODE_NEW);
-                        Matrix_Scale(var2, 0.15f, 1.0f, MTXMODE_APPLY);
+                    Matrix_Translate(entry->pos.x, entry->pos.y, 0.0f, MTXMODE_NEW);
+                    Matrix_Scale(var2, 0.15f, 1.0f, MTXMODE_APPLY);
 
-                        gDPSetPrimColor(OVERLAY_DISP++, 0, 0, entry->color.r, entry->color.g, entry->color.b,
-                                        (u8)alpha);
+                    gDPSetPrimColor(OVERLAY_DISP++, 0, 0, entry->color.r, entry->color.g, entry->color.b, (u8)alpha);
 
-                        Matrix_RotateZS((targetCtx->unk4B * 512), MTXMODE_APPLY);
+                    Matrix_RotateZS(targetCtx->rotation * 512, MTXMODE_APPLY);
 
-                        for (i = 0; i < 4; i++) {
-                            Matrix_RotateZS(0x4000, MTXMODE_APPLY);
-                            Matrix_Push();
-                            Matrix_Translate(entry->unkC, entry->unkC, 0.0f, MTXMODE_APPLY);
-                            gSPMatrix(OVERLAY_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
-                            gSPDisplayList(OVERLAY_DISP++, gZTargetLockOnTriangleDL);
-                            Matrix_Pop();
-                        }
+                    // Draw the 4 lock-on triangles
+                    for (lockOnIndex = 0; lockOnIndex < 4; lockOnIndex++) {
+                        Matrix_RotateZS(0x10000 / 4, MTXMODE_APPLY);
+                        Matrix_Push();
+                        Matrix_Translate(entry->distance, entry->distance, 0.0f, MTXMODE_APPLY);
+                        gSPMatrix(OVERLAY_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
+                        gSPDisplayList(OVERLAY_DISP++, gZTargetLockOnTriangleDL);
+                        Matrix_Pop();
                     }
+                }
 
-                    alpha -= 255 / 3;
-                    if (alpha < 0) {
-                        alpha = 0;
-                    }
+                alpha -= 255 / ARRAY_COUNT(targetCtx->lockOnEntries);
+                if (alpha < 0) {
+                    alpha = 0;
                 }
             }
         }
+    }
 
-        actor = targetCtx->unk_94;
-        if ((actor != NULL) && !(actor->flags & ACTOR_FLAG_CANT_LOCK_ON)) {
-            TatlColor* color = &sTatlColorList[actor->category];
+    actor = targetCtx->targetableOption;
```

Comment:
`targetableOption` makes me think of switch vs hold targeting. What is this actor doing?

Reply:
It is used for the Z target arrow which is drawn above an actor

Reply:
Then how does it differ from `arrowPointedActor`?

Reply:
Looking at how `gZTargetArrowDL` is drawn above a targets head, it seems `targetableOption` is the actor that's directly used for that, so maybe the name `arrowPointedActor` needs to be reconsidered since that's not the actor with the arrow pointed to it? Do you know what `arrowPointedActor` does? It has some role in the tatl actor and seems to be checked first before `targetableOption` in the main `z-targeting` player function


```c
actor = targetCtx->targetableOption;
if ((actor != NULL) && !(actor->flags & ACTOR_FLAG_CANT_LOCK_ON)) {
    TatlColor* color = &sTatlColorList[actor->category];

    POLY_XLU_DISP = Gfx_SetupDL(POLY_XLU_DISP, SETUPDL_7);

    Matrix_Translate(actor->focus.pos.x, actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y) + 17.0f,
                        actor->focus.pos.z, MTXMODE_NEW);
    Matrix_RotateYS(play->gameplayFrames * 0xBB8, MTXMODE_APPLY);
    Matrix_Scale((iREG(27) + 35) / 1000.0f, (iREG(28) + 60) / 1000.0f, (iREG(29) + 50) / 1000.0f, MTXMODE_APPLY);

    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, color->inner.r, color->inner.g, color->inner.b, 255);
    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_MODELVIEW | G_MTX_LOAD);
    gSPDisplayList(POLY_XLU_DISP++, gZTargetArrowDL);
}
```

Reply:
Sorry I missed this comment.
`arrowPointedActor` is very weird. It obviously isn't the actor that's being pointed by the arrow, but I don't really understand what it does either. If you comment the only part where it is being set to non-NULL, Tatl doesn't fly around the actor anymore and you can't lock-on anymore when pressing Z (you get everything else normal tho)

Reply:
The more I think about this the more I want to name `targetableOption` as `arrowPointedActor`, but that would mean leaving the current `arrowPointedActor` unnamed because I don't understand what it is used for

Reply:
Hmm, I think just flipping the two around works (or possibly even just `target` instead of `targetableOption` though that might be too general). You would expect a `targetableOption` to have Tatl fly towards it, so I think that makes sense. (If that does become `targetablOption` instead of just `target`, I would suggest changing `nextTarget` to `nextTargetableOption`)

Reply:
Yeah, hensldm's suggestion is alright, although I do want to mess around with this in game as well

Reply:
I'll swap those two members then